aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -2425,12 +2448,16 @@ def include_router(self, router: Router, prefix: str | None = None) -> None:
             # Middleware store the route without prefix, so we must not include prefix when grabbing
             middlewares = router._routes_with_middleware.get(route)
 
+            # Workaround to support backward-compatible interface
+            new_route = new_route[:-1]  # positional arguments until `middlewares` parameter
```

Comment:
I need some help with this one, maybe there is a better way.

If I understood correctly, `route` and `new_route` are really a `route_key` from `Router.route()` method. I must do this strange slicing to stay compatible with the current of `BaseRouter.route` abstract method:

```python
    @abstractmethod
    def route(
        self,
        rule: str,
        ...
        security: list[dict[str, list[str]]] | None = None,
        openapi_extensions: dict[str, Any] | None = None,
        # < -- cannot have a param here - someone may pass args by position
        middlewares: list[Callable[..., Any]] | None = None,
        # < -- must have param here
    ) -> Callable[[AnyCallableT], AnyCallableT]:
```

Reply:
Another idea would be to use something like [`typing.NamedTuple`](https://docs.python.org/3/library/typing.html#typing.NamedTuple) for the `route_key` variable and annotation for dictionary keys:

```python
from typing import NamedTuple

class RouteKeySomething(NamedTuple):
    rule: str
    methods: tuple[str]
    ...
    deprecated: bool
```

Reply:
Also added this to `BedrockAgentResolver` methods.